src/main/java/mate/academy/springboot/service/imlpl/UserServiceImpl.java

```diff
@@ -0,0 +1,34 @@
+package mate.academy.springboot.service.imlpl;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.springboot.dto.UserRegistrationRequestDto;
+import mate.academy.springboot.dto.UserResponseDto;
+import mate.academy.springboot.exception.RegistrationException;
+import mate.academy.springboot.mapper.UserMapper;
+import mate.academy.springboot.model.User;
+import mate.academy.springboot.repository.user.UserRepository;
+import mate.academy.springboot.service.UserService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
+            throw new RegistrationException("Can`t register user");
```

Comment:
```suggestion
            throw new RegistrationException("Can`t register user by email: ", email);
```
better make a message more informative